### ISSUE TITLE
VictoriaLogs Too Big columnsHeaderSize

### ISSUE THREAD

**Issue by jhodysetiawansekardono** [VictoriaMetrics/VictoriaMetrics#4834]
### Describe the bug

The bug seems to be related to the size of something called "columnsHeaderSize." It appears that this value has exceeded a specific limit. It indicates that the size of this component has exceeded the allowed limit of 8388608 bytes.

### To Reproduce

I don't know how to reproduce this issue, but I use VictoriaLogs to store logs from three Kubernetes clusters using Promtail. In the Promtail I use this configuration:
```
    server:
      log_level: info
      http_listen_port: 3101
    clients:
      - url: http://MY_VL_ADDRESS:9428/insert/loki/api/v1/push?_stream_fields=cluster,node_name,namespace,job,app,pod,container
    positions:
      filename: /run/promtail/positions.yaml
    scrape_configs:
      - job_name: kubernetes-pods
        pipeline_stages:
          - docker: {}
        kubernetes_sd_configs:
          - role: pod
        relabel_configs:
          - source_labels:
              - __meta_kubernetes_pod_controller_name
            regex: ([0-9a-z-.]+?)(-[0-9a-f]{8,10})?
            action: replace
            target_label: __tmp_controller_name
          - source_labels:
              - __meta_kubernetes_pod_label_app_kubernetes_io_name
              - __meta_kubernetes_pod_label_app
              - __tmp_controller_name
              - __meta_kubernetes_pod_name
            regex: ^;*([^;]+)(;.*)?$
            action: replace
            target_label: app
          - source_labels:
              - __meta_kubernetes_pod_label_app_kubernetes_io_instance
              - __meta_kubernetes_pod_label_release
            regex: ^;*([^;]+)(;.*)?$
            action: replace
            target_label: instance
          - source_labels:
              - __meta_kubernetes_pod_label_app_kubernetes_io_component
              - __meta_kubernetes_pod_label_component
            regex: ^;*([^;]+)(;.*)?$
            action: replace
            target_label: component
          - action: replace
            source_labels:
            - __meta_kubernetes_pod_node_name
            target_label: node_name
          - action: replace
            source_labels:
            - __meta_kubernetes_namespace
            target_label: namespace
          - action: replace
            replacement: $1
            separator: /
            source_labels:
            - namespace
            - app
            target_label: job
          - action: replace
            source_labels:
            - __meta_kubernetes_pod_name
            target_label: pod
          - action: replace
            source_labels:
            - __meta_kubernetes_pod_container_name
            target_label: container
          - action: replace
            replacement: /var/log/pods/*$1/*.log
            separator: /
            source_labels:
            - __meta_kubernetes_pod_uid
            - __meta_kubernetes_pod_container_name
            target_label: __path__
          - action: replace
            regex: true/(.*)
            replacement: /var/log/pods/*$1/*.log
            separator: /
            source_labels:
            - __meta_kubernetes_pod_annotationpresent_kubernetes_io_config_hash
            - __meta_kubernetes_pod_annotation_kubernetes_io_config_hash
            - __meta_kubernetes_pod_container_name
            target_label: __path__
          - target_label: cluster
            replacement: kubernetes-cluster-1
```

### Version

I use this version
```
victoria-logs-prod --version
victoria-logs-20230721-040437-tags-v0.3.0-victorialogs-0-gda0caa1d34
```

### Logs

```
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: 2023-08-12T10:27:38.890Z        panic        VictoriaMetrics/lib/logstorage/block.go:468        BUG: too big columnsHeaderSize: 8391854 bytes; mustn't exceed 8388608 bytes
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: panic: BUG: too big columnsHeaderSize: 8391854 bytes; mustn't exceed 8388608 bytes
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: goroutine 3650 [running]:
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:235 +0x65
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: panic({0xb79580, 0xc002c5df60})
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         runtime/panic.go:890 +0x263
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0xc44169, 0x5}, {0xc01b2c19f0, 0x4b}, 0x2?)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:281 +0xa96
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0xc44169, 0x5}, {0xc73a70?, 0xc001a69380?}, {0xc000496c78?, 0xc0022e0000?, 0xc00008b220?})
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0x1a5
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*block).mustWriteTo(0xc0068fe370, 0xc0042dc000, 0xc0004c42a0, 0xc001a69380)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block.go:468 +0x431
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*blockStreamWriter).mustWriteBlockInternal(0xc001a69380, 0xc0042dc000, 0xc0068fe370, 0xd50?)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block_stream_writer.go:269 +0x227
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*blockStreamWriter).MustWriteBlock(...)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block_stream_writer.go:252
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*blockStreamWriter).MustWriteRows(0x12a7420?, 0xc00007f680?, {0xc001c1ea80, 0x1, 0xd50}, {0xc0083a4000, 0x1, 0xc00})
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block_stream_writer.go:229 +0xea
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*inmemoryPart).mustInitFromRows(0xc00007f680, 0xc0004fec00)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/inmemory_part.go:76 +0x4e5
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*datadb).mustAddRows(0xc000374840, 0xc0004fec00)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/datadb.go:506 +0x7f
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*partition).mustAddRows(0xc00057bd80, 0xc0004fec00)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/partition.go:143 +0x325
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).MustAddRows(0xc00009c820, 0xc0004fec00)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage.go:400 +0x7fb
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage.MustAddRows(...)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage/main.go:79
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/insertutils.(*CommonParams).GetProcessLogMessageFunc.func1(0xc01dfec700?, {0xc01cd53000?, 0x0?, 0xc01d630780?})
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/insertutils/common_params.go:85 +0x8c
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/loki.parseProtobufRequest({0xc01da9e000, 0x6038d, 0x6c000}, 0xc00026c8a0)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/loki/loki_protobuf.go:97 +0x65a
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/loki.handleProtobuf(0xc01dfec600, {0xe24210, 0xc00041f9c0})
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/loki/loki_protobuf.go:43 +0x1fd
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/loki.RequestHandler({0xc01b357431?, 0x11?}, {0xe24210, 0xc00041f9c0}, 0xc01dfec600)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/loki/loki.go:31 +0xee
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert.RequestHandler({0xe24210, 0xc00041f9c0}, 0xc01dfec600)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/main.go:39 +0x21c
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: main.requestHandler({0xe24210?, 0xc00041f9c0}, 0xc01dfec600)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-logs/main.go:87 +0x2da
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc000111a50, {0xe24210?, 0xc00041f9c0}, 0xc01dfec600, 0xd73638)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:344 +0xd8d
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xe24210?, 0xc00041f9c0?}, 0xc0005a79b0?)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:196 +0x37
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: net/http.HandlerFunc.ServeHTTP(0xc12980?, {0xe24210?, 0xc00041f9c0?}, 0x4?)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         net/http/server.go:2122 +0x2f
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xe23c70, 0xc0000efea0}, 0xc0005a7ad0?)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         github.com/klauspost/compress@v1.16.7/gzhttp/compress.go:492 +0x4f2
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: net/http.HandlerFunc.ServeHTTP(0xc01b35743e?, {0xe23c70?, 0xc0000efea0?}, 0x6b9ad4?)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         net/http/server.go:2122 +0x2f
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: net/http.serverHandler.ServeHTTP({0xe21918?}, {0xe23c70, 0xc0000efea0}, 0xc01dfec600)
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         net/http/server.go:2936 +0x316
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: net/http.(*conn).serve(0xc002568630, {0xe24320, 0xc0002e1d40})
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         net/http/server.go:1995 +0x612
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]: created by net/http.(*Server).Serve
Aug 12 17:27:38 logserver victoria-logs-prod[2063119]:         net/http/server.go:3089 +0x5ed
Aug 12 17:27:38 logserver systemd[1]: victorialogs.service: Main process exited, code=exited, status=1/FAILURE
```

### Used command-line flags

```
victoria-logs-prod \
-storageDataPath=/mnt/data/victorialogs \
-retentionPeriod=1y \
-insert.maxLineSizeBytes=1GB
```

**Comment by valyala** [comment]
@jhodysetiawansekardono , thanks for filing the issue!

It looks like Loki ingests too long log message to VictoriaLogs, with the length exceeding 8Mb.

VictoriaLogs doesn't check for the length of input log messages, so it results in this panic. I'm going add an ability to check for log message lengths during data ingestion - this will substitute the panic with the `too long log line` error log message and the reject of such long log messages.

Temporary workaround is to specify `max_line_size` and `max_line_size_truncate` options at [Loki config](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#limits_config). The `max_line_size` option must be set to value smaller than 8Mb (ideally, it shouldn't exceed a few tens of Kb, since longer log messages usually point to some improper logging).

**Comment by victrola0000** [comment]
We are seeing a similar problem with Logstash sending data to VictoriaLogs.  Not sure if the fix above will help in this matter, or if this may help provide a different fix type.  

After I had restarted logstash and then victoria-logs the issue went away, most likely the offending log line was dropped.

**Version**
1.93.0
Running via Docker Run

**Command Line Options**
```
-insert.maxLineSizeBytes=10485760
-retentionPeriod=1y
-memory.allowedPercent=75
```

**Output**

> 2023-08-18T08:17:12.174Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: victoria-logs-20230812-215826-tags-v1.93.0-0-g59f7d810c9
> 2023-08-18T08:17:12.174Z        info    VictoriaMetrics/lib/logger/flag.go:13   command-line flags
> 2023-08-18T08:17:12.174Z        info    VictoriaMetrics/lib/logger/flag.go:20     -insert.maxLineSizeBytes="10485760"
> 2023-08-18T08:17:12.174Z        info    VictoriaMetrics/lib/logger/flag.go:20     -memory.allowedPercent="75"
> 2023-08-18T08:17:12.174Z        info    VictoriaMetrics/lib/logger/flag.go:20     -retentionPeriod="1y"
> 2023-08-18T08:17:12.174Z        info    VictoriaMetrics/app/victoria-logs/main.go:42    starting VictoriaLogs at ":9428"...
> 2023-08-18T08:17:12.175Z        info    VictoriaMetrics/app/vlstorage/main.go:52        opening storage at -storageDataPath=victoria-logs-data
> 2023-08-18T08:17:12.175Z        info    VictoriaMetrics/lib/memory/memory.go:42 limiting caches to 25230766080 bytes, leaving 8410255360 bytes to the OS according to -memory.allowedPercent=75
> 2023-08-18T08:17:14.014Z        info    VictoriaMetrics/app/vlstorage/main.go:58        successfully opened storage in 1.839 seconds; partsCount: 1092; blocksCount: 2381757; rowsCount: 1763886511; sizeBytes: 157927952035
> 2023-08-18T08:17:14.014Z        info    VictoriaMetrics/app/victoria-logs/main.go:50    started VictoriaLogs in 1.839 seconds; see https://docs.victoriametrics.com/VictoriaLogs/
> 2023-08-18T08:17:14.014Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:96 starting http server at http://127.0.0.1:9428/
> 2023-08-18T08:17:14.014Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:97 pprof handlers are exposed at http://127.0.0.1:9428/debug/pprof/
> 2023-08-18T08:17:20.130Z        panic   VictoriaMetrics/lib/logstorage/block.go:468     BUG: too big columnsHeaderSize: 8695954 bytes; mustn't exceed 8388608 bytes
> panic: BUG: too big columnsHeaderSize: 8695954 bytes; mustn't exceed 8388608 bytes
> 
> goroutine 470 [running]:
> github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:235 +0x5b
> panic({0x88e620?, 0xc000196030?})
>         runtime/panic.go:920 +0x270
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0x900916, 0x5}, {0xc0006d8050, 0x4b}, 0x2?)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:281 +0xa91
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0x900916, 0x5}, {0x91eb81?, 0xc000458680?}, {0xc00066abc0?, 0x6e0f27?, 0xc0000a10a8?})
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0x194
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*block).mustWriteTo(0xc000405c20, 0xc0000fc108, 0xc0000a1080, 0xc000458680)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block.go:468 +0x43e
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*blockStreamWriter).mustWriteBlockInternal(0xc000458680, 0xc0000fc108, 0xc000405c20, 0x2?)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block_stream_writer.go:269 +0x212
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*blockStreamWriter).MustWriteBlock(...)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block_stream_writer.go:252
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*blockStreamWriter).MustWriteRows(0xc000391e90?, 0xc000b3e4b0?, {0xc000a1a0a0, 0x1, 0x2}, {0xc000391e90, 0x1, 0x2})
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/block_stream_writer.go:229 +0xe5
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*inmemoryPart).mustInitFromRows(0xc000b3e4b0, 0xc00082a300)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/inmemory_part.go:66 +0x205
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*datadb).mustAddRows(0xc000774f20, 0xc00082a300)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/datadb.go:506 +0x73
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*partition).mustAddRows(0xc0001b8940, 0xc00082a300)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/partition.go:143 +0x310
> github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).MustAddRows(0xc00010a140, 0xc00082a240)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage.go:442 +0x790
> github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage.MustAddRows(0xc00082a240?)
>         github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage/main.go:79 +0x25
> github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch.RequestHandler.(*CommonParams).GetProcessLogMessageFunc.func1(0xc0002920ad?, {0xc0000d6800?, 0xc0000d6800?, 0x15?})
>         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/insertutils/common_params.go:85 +0x75
> github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch.readBulkLine(0xc00066b480, {0xc0002920ad, 0xa}, {0xc0002920e9, 0x7}, 0xc00066b610)
>         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch/elasticsearch.go:215 +0x5ab
> github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch.readBulkRequest({0x7fe95293f1d0?, 0xc0008144c0?}, 0x86?, {0xc0002920ad, 0xa}, {0xc0002920e9, 0x7}, 0x7fffffffffffffff?)
>         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch/elasticsearch.go:149 +0x36f
> github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch.RequestHandler({0xc00029209a, 0x6}, {0xa8ed40, 0xc0000ae000}, 0xc0001cc300)
>         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/elasticsearch/elasticsearch.go:99 +0x5ad
> github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert.RequestHandler({0xa8ed40, 0xc0000ae000}, 0xc0001cc300)
>         github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/main.go:36 +0x196
> main.requestHandler({0xa8ed40?, 0xc0000ae000}, 0xc0001cc300)
>         github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-logs/main.go:87 +0x2cc
> github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc0001966c0, {0xa8ed40?, 0xc0000ae000}, 0xc0001cc300, 0xa19d80)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:344 +0xd8d
> github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xa8ed40?, 0xc0000ae000?}, 0xc0002ce0e0?)
>         github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:196 +0x31
> net/http.HandlerFunc.ServeHTTP(0xce3ee0?, {0xa8ed40?, 0xc0000ae000?}, 0x4?)
>         net/http/server.go:2136 +0x29
> github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xa8e290, 0xc0002ce0e0}, 0xc0001420b0?)
>         github.com/klauspost/compress@v1.16.7/gzhttp/compress.go:486 +0x357
> net/http.HandlerFunc.ServeHTTP(0x40e4e5?, {0xa8e290?, 0xc0002ce0e0?}, 0xc0002ce001?)
>         net/http/server.go:2136 +0x29
> net/http.serverHandler.ServeHTTP({0xa8d460?}, {0xa8e290?, 0xc0002ce0e0?}, 0x6?)
>         net/http/server.go:2938 +0x8e
> net/http.(*conn).serve(0xc0005fe1b0, {0xa8f9f8, 0xc0003ba330})
>         net/http/server.go:2009 +0x5f4
> created by net/http.(*Server).Serve in goroutine 52
>         net/http/server.go:3086 +0x5cb

**Comment by jhodysetiawansekardono** [comment]
> @jhodysetiawansekardono , thanks for filing the issue!
> 
> It looks like Loki ingests too long log message to VictoriaLogs, with the length exceeding 8Mb.
> 
> VictoriaLogs doesn't check for the length of input log messages, so it results in this panic. I'm going add an ability to check for log message lengths during data ingestion - this will substitute the panic with the `too long log line` error log message and the reject of such long log messages.
> 
> Temporary workaround is to specify `max_line_size` and `max_line_size_truncate` options at [Loki config](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#limits_config). The `max_line_size` option must be set to value smaller than 8Mb (ideally, it shouldn't exceed a few tens of Kb, since longer log messages usually point to some improper logging).

Thanks for suggesting that using the temporary workaround by specifying the `max_line_size` and `max_line_size_truncate` options in the Loki config is a valuable tip for anyone encountering this issue.

**Comment by valyala** [comment]
@jhodysetiawansekardono , @victrola0000 , could you try building VictoriaLogs from the commit 8dce4eb18998f06b1f2d15e25dad7afd9345e8a0 according to [these docs](https://docs.victoriametrics.com/VictoriaLogs/QuickStart.html#building-from-source-code) and verifying whether it is free from the `too big columnsHeaderSize` error, which could appear during data ingestion? This commit will be included in the next release of VictoriaLogs if it doesn't introduce now issues.

**Comment by jhodysetiawansekardono** [comment]
@valyala, I'll let you know if this issue still occurs in the v0.4.0.

**Comment by valyala** [comment]
FYI, it has been appeared that the `v0.4.0-victorialogs` contained [the issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5112), which could prevent from normal operation. The issue has been fixed in [v0.4.1-victorialogs](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v0.4.1-victorialogs).

**Comment by valyala** [comment]
Closing the issue as fixed. @jhodysetiawansekardono , feel free re-opening it if this issue still present in [v0.4.1-victorialogs](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v0.4.1-victorialogs) or in newer releases.
